pkg/cmd/pr/create/regex_writer.go

```diff
@@ -0,0 +1,46 @@
+package create
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"regexp"
+)
+
+func NewRegexWriter(out io.Writer, regexp *regexp.Regexp, repl string) io.Writer {
+	return &RegexWriter{out: out, regexp: *regexp, repl: repl}
+}
+
+type RegexWriter struct {
+	out    io.Writer
+	regexp regexp.Regexp
+	repl   string
+}
+
+func (s RegexWriter) Write(data []byte) (int, error) {
+	filtered := []byte{}
+	repl := []byte(s.repl)
+	scanner := bufio.NewScanner(bytes.NewReader(data))
+
+	for scanner.Scan() {
+		b := scanner.Bytes()
+		f := s.regexp.ReplaceAll(b, repl)
+		if len(f) > 0 {
+			filtered = append(filtered, f...)
+			filtered = append(filtered, []byte("\n")...)
```

Comment:
This appends a newline after every call to `Scan()`, including the final one at the end of reading `data`, but `Write`s are not guaranteed to always be aligned with the newline boundary, and so the original `data` might not have had a terminating newline. As a result, this function can result in superfluous newlines being written. I'm not sure how to solve this, but extra care should be taken not to mess up the valid output of `git push`.